parser: allow optional = for SURVIVE x FAILURE

otan opened the pull request:
Release note (sql change): Allow an optional `=` character for SURVIVE,
e.g. `ALTER DATABASE d SURVIVE = ZONE FAILURE`.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/56881)
<!-- Reviewable:end -->

otan committed:
parser: allow optional = for SURVIVE x FAILURE

Release note (sql change): Allow an optional `=` character for SURVIVE,
e.g. `ALTER DATABASE d SURVIVE = ZONE FAILURE`.

ajstorm reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/56881#-:-MMWD79F90dHrYJxsO0Y:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @ajstorm)



<!-- Sent from Reviewable.io -->

otan commented:
bors r=ajstorm
